### ISSUE TITLE
Dependency issue

### ISSUE THREAD

**Issue by zalky** [s3-wagon-private/s3-wagon-private#38]
I noticed there was a side discussion on dependencies in [this issue](https://github.com/s3-wagon-private/s3-wagon-private/issues/30), so apologies if this is a dupe.

I seem to be experiencing a dependency conflict with the latest alpha2, but I can't quit seem to figure out why this is happening. I've excluded `jackson-core` and `jackson-databind`, as well as `aws-java-sdk` from all the relevant dependencies in my own project. Using boot.
```
Exception in thread "pool-5-thread-1" java.lang.NoSuchMethodError: com.fasterxml.jackson.core.JsonParser.setCurrentValue(Ljava/lang/Object;)V                          
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:243)                                                         
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:136)                                                                
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3564)                                                                        
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2533)                                                                               
        at com.amazonaws.internal.config.InternalConfig.loadfrom(InternalConfig.java:236)                                                                              
        at com.amazonaws.internal.config.InternalConfig.load(InternalConfig.java:249)                                                                                  
        at com.amazonaws.internal.config.InternalConfig$Factory.<clinit>(InternalConfig.java:304)                                                                      
        at com.amazonaws.util.VersionInfoUtils.userAgent(VersionInfoUtils.java:141)
        at com.amazonaws.util.VersionInfoUtils.initializeUserAgent(VersionInfoUtils.java:136)                                                                          
        at com.amazonaws.util.VersionInfoUtils.getUserAgent(VersionInfoUtils.java:97)                                                                                  
        at com.amazonaws.ClientConfiguration.<clinit>(ClientConfiguration.java:60)
        at org.springframework.build.aws.maven.S3Utils.getClientConfiguration(S3Utils.java:45)                                                                         
        at org.springframework.build.aws.maven.PrivateS3Wagon.connectToRepository(PrivateS3Wagon.java:92)                                                              
        at org.springframework.build.aws.maven.AbstractWagon.connect(AbstractWagon.java:143)                                                                           
        at org.sonatype.aether.connector.wagon.WagonRepositoryConnector.connectWagon(WagonRepositoryConnector.java:345)                                                
        at org.sonatype.aether.connector.wagon.WagonRepositoryConnector.pollWagon(WagonRepositoryConnector.java:385)                                                   
        at org.sonatype.aether.connector.wagon.WagonRepositoryConnector$GetTask.run(WagonRepositoryConnector.java:571)                                                 
        at org.sonatype.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:60)                                                           
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)                                                                             
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)                                                                             
        at java.lang.Thread.run(Thread.java:745)
```

Any ideas? Is there anything I can do on my end to solve this?

**Comment by zalky** [comment]
Thanks @danielcompton for the response. Is there a conventional way to check the dependencies in an uberjar?

I'm using boot `2.5.5`. Looking through the source it seems it is pulling in cheshire `5.3.1`, which in turn is pulling in jackson `2.3.1`.

**Comment by danielcompton** [comment]
> Thanks @danielcompton for the response. Is there a conventional way to check the dependencies in an uberjar?

Not really sure apart from opening the JAR up, don't think there is.

> I'm using boot 2.5.5. Looking through the source it seems it is pulling in cheshire 5.3.1, which in turn is pulling in jackson 2.3.1.

Yeah that'll be the problem. It looks like there's a newer version of Cheshire out which has a more recent Jackson version (although it's newer than the version that we bundle, so not sure what that would do).

**Comment by sheelc** [comment]
On a perhaps related note, we had issues with both earlier and later versions of Jackson outside of 2.5.5. There's some context from this [PR](https://github.com/s3-wagon-private/s3-wagon-private/pull/33) where we worked through similar sounding "No such method" exceptions as you've experienced.

**Comment by peterhhchan** [comment]
@zalky Quick note:  I managed to "fix" this error by compiling a version of boot where I updated the version of Cheshire to 5.7.0.  However, I am now experiencing #40.

**Comment by peterhhchan** [comment]
Quick update, I managed to get everything working by changing the dependency in `boot/aether` from `org.apache.maven.wagon/wagon-http "2.9"` to "2.12".

**Comment by kennyjwilli** [comment]
Yup, this issue is blocking me from upgrading to 1.3.0.

**Comment by zalky** [comment]
Should be fixed with #59 and boot-clj/boot#679
